core/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -787,7 +797,21 @@ public void writeTo(StreamOutput out) throws IOException {
             nodes.writeTo(out);
             metaData.writeTo(out);
             blocks.writeTo(out);
-            customs.writeTo(out);
+            Diff<ImmutableOpenMap<String, Custom>> customsDiff = customs;
+            if (out.getVersion().before(SnapshotDeletionsInProgress.VERSION_INTRODUCED)) {
+                customsDiff = removeSnapshotDeletionsCustomDiff(customsDiff);
```

Comment:
Sounds good, when #22336 is in, we can go over improvements to the custom diff logic here.